ServerCore/Pages/Threads/PuzzleThread.cshtml.cs

```diff
@@ -74,7 +74,27 @@ public async Task<IActionResult> OnGetAsync(int? puzzleId, int? teamId, int? pla
                 return NotFound();
             }
 
-            Team team = null;
+            // If we are showing hints, only allow the thread to be seen if the user has unlocked it
+            if (!Event.HideHints)
+            {
+                string returnUrl = HttpContext.Request.Path.HasValue ? HttpContext.Request.Path.Value : null;
+                if (Event.DefaultCostForHelpThread < 0)
+                {
+                    return RedirectToPage("/ErrorPage", new { ErrorMessage = "Help threads are not allowed for this event" });
```

Comment:
Is this something that can happen without the player customizing the URL? If not, then I think `return Forbid();` is enough.

Reply:
Same goes for the redirect on line 93.

Reply:
done